Week2/TriplateSum.java

```diff
@@ -0,0 +1,81 @@
+package sum3;
+
+import java.util.*;
+
+public class TriplateSum 
+{
+	public static Set<Set<Integer>> Sum3(int ar[])
+	{
+		 Set<Set<Integer>> bags = new HashSet<>();
+		 int ans[]= {0},n=ar.length,left,right;
+		 if(ar == null || ar.length<3) //array is NULL or less than two elements;
+			 return null;
+		 Arrays.sort(ar); //sorting array
+		 for(int i=0;i<n-2;i++) //traversing array for top n-2 elements 
+		 {
+			 left=i+1; //next bigger element from left
+			 right=n-1;// most largest element
+		 
+		  while(left<right) //till elements indices don't coincide

```

Comment:
Spacing inconsistency - this while loop looks like its indented at the same scope as the for loop when really it is inside it. That can be confusing to the reader